torch/csrc/jit/fuser/compiler.cpp

```diff
@@ -104,7 +107,8 @@ static std::vector<int64_t> getInputDependencies(const Value* output) {
     const Value* val = queue.back();
     queue.pop_back();
     const Node* producer = val->node();
-    if (producer->kind() == prim::Param) {
+    if (producer->kind() == prim::Param &&
```

Comment:
Maybe add a big comment here that non-inputs must be sizes that are not used for the output computation (and only used for gradient computation)